packages/content-req-app/src/app/pr-list/pr-list.component.ts

```diff
@@ -64,14 +65,14 @@ export class PrListComponent implements OnInit, PluginComponent {
   }
 
   initialize(context: PluginContext): void {
-    this.appEventPub = context.publisher('app:event');
+    this.appEventPub = context.messaging.createPublisher('app:event', EventType.ONLY_LAST);
   }
 
   getConfiguration(): Partial<PluginConfiguration> {
     return new PrListPluginConfiguration();
   }
 
-  onItemClicked($event: MouseEvent, id: string) {
+  onItemClicked($event: MouseEvent, id: string): void {
     this.appEventPub.publish(new TextMessage('Hello from Pr List: ' + id))
```

Comment:
As I commented above. I just want to hit publish and it shoudl create a channel on demand, wihtout creating Publisher. 

This is why I have Topics and I use to defineTopic which has some specific params so I dont need to pass event Type into subscriber. I can pull it out from Topic definition. 

I will check the app-shell.

Reply:
Answered in your first comment, it's your code. Not yet refactored